### PR TITLE
added Exp6-AAD, AMBER-AA, TRAPPE-UA, Mie-UA, FF styles tests and XML …

### PR DESCRIPTION
**Pull request by bc118**
##Added Exp6-AAD, AMBER-AA, TRAPPE-UA, Mie-UA, FF styles tests and XML FF examples in the test/files directory.  Changed the FF form to U_opls... This resolves the issue below for this repo, but a few tests examples, will be created here https://github.com/GOMC-WSU/GOMC_Examples/tree/main/MoSDeF-dihedral-fit for the Exp6, Mie, TRAPPE, and Amber FF examples.

https://github.com/GOMC-WSU/MoSDeF-dihedral-fit/issues/99

## Changed the FF form to U_opls and periodic dihedrals, RB-torsion,  This resolves this part of the following issue: 
issue = https://github.com/GOMC-WSU/MoSDeF-dihedral-fit/issues/100
 
Line 100-103: I would like to suggest for the equations to use physical meaningful symbols as targets quantities (U or V). Examples: V_{dihedral}^{OPLS}(\theta) (instead of OPLS_{Energy}), V_{dihedral}^{RB}(\psi), V_{dihedral}^{Periodic}(\theta)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,7 +23,8 @@ fit_dihedral_with_gomc(
         atom_type_naming_style='general',
         gomc_cpu_cores=1,
         r_squared_min=0.99,
-        r_squared_rtol=1e-03
+        r_squared_rtol=1e-03,
+        opls_set_k0_zero=True
```
Comment (daico007) [inline README.md:27]:
```suggestion
        opls_force_k0_zero=True
```

Comment (daico007) [reply README.md:27]: or we can change the rest to `set` (tbh I like `set` better)

File: mosdef_dihedral_fit/dihedral_fit/fit_dihedral_with_gomc.py [hunk 2]
```diff
@@ -716,6 +740,12 @@ def fit_dihedral_with_gomc(
             f"ERROR: The 'qm_engine' is a {type(qm_engine)}, but it needs to be a str."
         )
 
+    # check if opls_force_k0_zero' is a bool
+    if not isinstance(opls_force_k0_zero, bool):
+        raise TypeError(
+            "ERROR: Please enter the 'opls_force_k0_zero' file as a bool."
```
Comment (daico007) [inline mosdef_dihedral_fit/dihedral_fit/fit_dihedral_with_gomc.py:746]:
```suggestion
            "ERROR: Please enter the 'opls_force_k0_zero' as a bool."
```

File: mosdef_dihedral_fit/dihedral_fit/fit_dihedral_with_gomc.py [hunk 3]
```diff
@@ -764,10 +794,8 @@ def fit_dihedral_with_gomc(
 
     # liquid_box_0_length_nm must be a value <= 999.8 nm and an interger value in angstroms <= 9998 Ang
     liquid_box_0_length_nm = 999.8
-    print(
-        "INFO: Started Building the fragment for the GOMC simulation with dihedral k values = 0"
-    )
 
+    # Started Building the fragment for the GOMC simulation with dihedral k values = 0"
```
Comment (daico007) [inline mosdef_dihedral_fit/dihedral_fit/fit_dihedral_with_gomc.py:798]:
```suggestion
    # Started building the fragment for the GOMC simulation with dihedral k values = 0
```

File: mosdef_dihedral_fit/dihedral_fit/fit_dihedral_with_gomc.py [hunk 4]
```diff
@@ -779,13 +807,7 @@ def fit_dihedral_with_gomc(
...
     )
 
-    print(
-        "INFO: Finished Building the fragment for the GOMC simulation with dihedral k values = 0"
-    )
-
-    print(
-        "INFO: Started Building the Charmm Object for the GOMC simulation with dihedral k values = 0"
-    )
+    # Started Building the Charmm Object for the GOMC simulation with dihedral k values = 0"
```
Comment (daico007) [inline mosdef_dihedral_fit/dihedral_fit/fit_dihedral_with_gomc.py:810]:
```suggestion
    # Started building the Charmm object for the GOMC simulation with dihedral k values = 0
```

File: mosdef_dihedral_fit/dihedral_fit/fit_dihedral_with_gomc.py [hunk 5]
```diff
@@ -801,22 +823,13 @@ def fit_dihedral_with_gomc(
...
     )
-    print(
-        "INFO: Finished Building the Charmm Object for the GOMC simulation with dihedral k values = 0"
-    )
 
-    # Write the write the FF (.inp), psf, pdb, and GOMC control files
-    print(
-        "INFO: Started Writing the PDB, PSF, and FF files for the GOMC simulation with dihedral k values = 0"
-    )
+    # Started Writing the PDB, PSF, and FF files for the GOMC simulation with dihedral k values = 0"
```
Comment (daico007) [inline mosdef_dihedral_fit/dihedral_fit/fit_dihedral_with_gomc.py:827]:
```suggestion
    # Started writing the PDB, PSF, and FF files for the GOMC simulation with dihedral k values = 0
```

File: mosdef_dihedral_fit/dihedral_fit/fit_dihedral_with_gomc.py [hunk 6]
```diff
@@ -931,20 +944,14 @@ def fit_dihedral_with_gomc(
         CrankShaftFreq = 0
         SwapFreq = 0
 
-        print("#**********************")
-        print(
-            "Started: NVT GOMC control file writing for the GOMC simulation with dihedral k values = 0"
-        )
-        print("#**********************")
+        # NVT GOMC control file writing for the GOMC simulation with dihedral k values = 0"
```
Comment (daico007) [inline mosdef_dihedral_fit/dihedral_fit/fit_dihedral_with_gomc.py:947]:
```suggestion
        # NVT GOMC control file writing for the GOMC simulation with dihedral k values = 0
```

File: mosdef_dihedral_fit/dihedral_fit/fit_dihedral_with_gomc.py [hunk 7]
```diff
@@ -1646,6 +1610,29 @@ def fit_dihedral_with_gomc(
...
+    # which is changed to 'k0_forced_to_0_sorted_const_1_minus_Cos_0_phi_data_lists'.
+    # If 'sorted_const_1_minus_Cos_0_phi_data_lists' = all 1s, then k0=constant,
+    # which is changed to 'k0_is_constant_sorted_const_1_minus_Cos_0_phi_data_lists'.
+    # It is critical that 'const_1_minus_Cos_0_phi_data' be all (0, 0, .., 0) for k0=0
+    # and all (1, 1, .., 1) 'const_1_minus_Cos_0_phi_data' for k0=constant.
+    k0_forced_to_0_list = []
+    k0_is_constant_list = []
+    for determine_k0_m in range(
+        0, len(sorted_const_1_minus_Cos_0_phi_data_lists)
+    ):
```
Comment (daico007) [inline mosdef_dihedral_fit/dihedral_fit/fit_dihedral_with_gomc.py:1625]:
```suggestion
    for determine_k0_m in range(len(sorted_const_1_minus_Cos_0_phi_data_lists)):
```

**Review by daico007 — APPROVED** [review]
everything looks fine on quick skim through, only some minor typo

File: README.md (hunk 1 above)
Comment (bc118) [reply README.md:27]: I had that but I changed to force.  I thought force was stronger because it does not allow for later shifting that is standard

Comment (daico007) [reply README.md:27]: then `force` sounds reasonable then

Comment (bc118) [reply README.md:27]: discussing with everyone, I think we will leave at `opls_force_k0_zero`

**Pull request merged**